src/display/text_layer.js

```diff
@@ -213,19 +531,85 @@ var renderTextLayer = (function renderTextLayerClosure() {
       var styles = this._textContent.styles;
       var textDivs = this._textDivs;
       var viewport = this._viewport;
+      var bounds = [];
+      var enhanceTextSelection = this._enhanceTextSelection;
+      var enhanceOnClick = this._enhanceOnClick;
       for (var i = 0, len = textItems.length; i < len; i++) {
-        appendText(textDivs, viewport, textItems[i], styles);
+        appendText(textDivs, viewport, textItems[i], styles, bounds,
+                   enhanceTextSelection);
+      }
+      if (enhanceTextSelection) {
+        expand(bounds, this._viewport);
+        if (enhanceOnClick) {
+          this._bindMouse(textDivs);
+        }
       }
-
       if (!timeout) { // Render right away
-        render(this);
+        render(this, enhanceOnClick);
       } else { // Schedule
         var self = this;
         this._renderTimer = setTimeout(function() {
-          render(self);
+          render(self, enhanceOnClick);
           self._renderTimer = null;
         }, timeout);
       }
+    },
+
+    _bindMouse: function TextLayer_bindMouse(textDivs) {
```

Comment:
Also, I don't think that you need to pass `textDivs` to this method. You should be able to use `this._textDivs` instead.